examples/plot_clip.py

```diff
@@ -0,0 +1,117 @@
+"""
+Clip Vector Data with GeoPandas
+==================================================================
+
+Learn how to clip geometries to the boundary of a polygon geometry
+using GeoPandas.
+
+"""
+
+###############################################################################
+# Clip Vector Data in Python Using GeoPandas
+# ---------------------------------------------
+#
+# The example below shows you how to clip a set of vector geometries
+# to the spatial extent / shape of another vector object. Both sets of geometries
+# must be opened with GeoPandas as GeoDataFrames and be in the same Coordinate
+# Reference System (CRS) for the ``clip()`` function in GeoPandas to work.
+#
+# This example uses GeoPandas example data ``'naturalearth_cities'`` and
+# ``'naturalearth_lowres'``, alongside a custom rectangle geometry made with
+# shapely and then turned into a GeoDataFrame.
+#
+# .. note::
+#    The object to be clipped will be clipped to the full extent of the clip
+#    object. If there are multiple polygons in clip object, the input data will
+#    be clipped to the total boundary of all polygons in clip object.
+
+###############################################################################
+# Import Packages
+# ---------------
+#
+# To begin, import the needed packages.
+
+import matplotlib.pyplot as plt
+import geopandas
+import geopandas.clip as gc
```

Comment:
@jorisvandenbossche this implementation worked well!! do you want clip to live in the tools directory? @nkorinek did attempt to move it into tools but noticed for some reason it didn't register properly from the tools directory. Many thanks.

Reply:
@lwasser If you add `from geopandas.tools import clip  # noqa` to `geopandas/__init__.py` it should work. It should definitely be in the `tools` dir.

Reply:
AHHHH that makes sense. ok that's the path specification within the package. Thanks @martinfleis we will try that!

Reply:
@martinfleis i just pushed a commit that implemented this move. we had to import it as `geopandas.tools.clip import clip` . This is a learning moment for me. SJOIN seems to import without the additional module name in geopandas. Is that because there is only one function within SJOIN whereas clip has a few helper functions and the main clip function? 

OR are we missing a step? thank you!

Reply:
Oh yes. You have to update `__init__.py` in `tools` as well. `from .clip import clip` and add `clip` to `__all__`.

Reply:
OK i suspected something was missing but i wasn't sure what that was. All is working now as expected. THANK YOU again @martinfleis !!